dubinc/dub
# Implement Dub CLI OAuth with PKCE Login

Feature #1334

### Steps to Test Locally:

Create an OAuth2 Application in Dub:

- Navigate to the [OAuth Apps tab](https://app.dub.co/settings/oauth-apps) in your Dub workspace.
- Click **`Create OAuth App`**.
- Fill out the required fields to set up your OAuth2 application.

Enable PKCE:

- Make sure to turn on the PKCE option in your OAuth2 application settings.

 Update Client ID:

```javascript
const oauth2Client = new OAuth2Client({
  // TODO: add client id here
  clientId: "",
  authorizationEndpoint: "https://app.dub.co/oauth/authorize",
  tokenEndpoint: "https://api.dub.co/oauth/token",
});
```

That's it!

You can now test the OAuth2 flow locally by logging into the CLI.
